src/main/java/net/sf/jabref/model/database/DuplicationChecker.java

```diff
@@ -64,38 +69,45 @@ public int getNumberOfKeyOccurrences(String key) {
 
     }
 
-    //========================================================
-    // keep track of all the keys to warn if there are duplicates
-    //========================================================
-    public boolean addKeyToSet(String key) {
+    /**
+     * If the given key is null or the key is empty, then false will be returned and the key will not be added.
```

Comment:
Try to write a level above this level.

Helper function for counting the number of the key usages. Adds the given key to the internal keyset together with the count of it. The counter is increased if the key already exists, otherwise set to 1. Special case: If a null or empty key is passed, it is not counted and thus not added.